string source = rogr
string target = cameraturret
time timestamp
